crates/bevy_ecs/macros/src/component.rs

```diff
@@ -116,42 +159,64 @@ fn parse_component_attr(ast: &DeriveInput) -> Result<Attrs> {
-                attrs.storage = match nested.value()?.parse::<LitStr>()?.value() {
-                    s if s == TABLE => StorageTy::Table,
-                    s if s == SPARSE_SET => StorageTy::SparseSet,
-                    s => {
-                        return Err(nested.error(format!(
-                            "Invalid storage type `{s}`, expected '{TABLE}' or '{SPARSE_SET}'.",
-                        )));
-                    }
-                };
-                Ok(())
-            } else if nested.path.is_ident(ON_ADD) {
-                attrs.on_add = Some(nested.value()?.parse::<ExprPath>()?);
-                Ok(())
-            } else if nested.path.is_ident(ON_INSERT) {
-                attrs.on_insert = Some(nested.value()?.parse::<ExprPath>()?);
-                Ok(())
-            } else if nested.path.is_ident(ON_REPLACE) {
-                attrs.on_replace = Some(nested.value()?.parse::<ExprPath>()?);
-                Ok(())
-            } else if nested.path.is_ident(ON_REMOVE) {
-                attrs.on_remove = Some(nested.value()?.parse::<ExprPath>()?);
-                Ok(())
-            } else {
-                Err(nested.error("Unsupported attribute"))
-            }
-        })?;
+    for attr in ast.attrs.iter() {
+        if attr.path().is_ident(COMPONENT) {
+            attr.parse_nested_meta(|nested| {
+                if nested.path.is_ident(STORAGE) {
+                    attrs.storage = match nested.value()?.parse::<LitStr>()?.value() {
+                        s if s == TABLE => StorageTy::Table,
+                        s if s == SPARSE_SET => StorageTy::SparseSet,
+                        s => {
+                            return Err(nested.error(format!(
+                                "Invalid storage type `{s}`, expected '{TABLE}' or '{SPARSE_SET}'.",
+                            )));
+                        }
+                    };
+                    Ok(())
+                } else if nested.path.is_ident(ON_ADD) {
+                    attrs.on_add = Some(nested.value()?.parse::<ExprPath>()?);
+                    Ok(())
+                } else if nested.path.is_ident(ON_INSERT) {
+                    attrs.on_insert = Some(nested.value()?.parse::<ExprPath>()?);
+                    Ok(())
+                } else if nested.path.is_ident(ON_REPLACE) {
+                    attrs.on_replace = Some(nested.value()?.parse::<ExprPath>()?);
+                    Ok(())
+                } else if nested.path.is_ident(ON_REMOVE) {
+                    attrs.on_remove = Some(nested.value()?.parse::<ExprPath>()?);
+                    Ok(())
+                } else {
+                    Err(nested.error("Unsupported attribute"))
+                }
+            })?;
+        } else if attr.path().is_ident(REQUIRE) {
+            let punctuated =
+                attr.parse_args_with(Punctuated::<Require, Comma>::parse_terminated)?;
+            attrs.requires = Some(punctuated);
+        }
```

Comment:
It seems this will silently ignore repeated `requires` attributes. IMO this should instead either error on multiple `requires` or merge them into one.

Reply:
I'm in favor of them being merged, since that would make life easier for macros that automatically generate the `#[requires(...)]` attribute.

Reply:
Merging makes sense to me!